pkg/debug/start_debug.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package debug
+
+import (
+	ctx "context"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	appsv1 "k8s.io/api/apps/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func StartDebug(context *k8sutil.Context, clusterNamespace, deploymentName, alternateImageValue string) {
+	deployment, err := verifyDeploymentExists(context, clusterNamespace, deploymentName)
+	if err != nil {
+		fmt.Printf("Missing mon or osd deployment name %s. %v\n", deploymentName, err)
+		os.Exit(1)
+	}
+
+	if alternateImageValue != "" {
+		deployment.Spec.Template.Spec.Containers[0].Image = alternateImageValue
+	}
+
+	labels := deployment.Labels
+	labels["ceph.rook.io/do-not-reconcile"] = "true"
+
+	deployment.Spec.Template.Spec.Containers[0].LivenessProbe = nil
+	deployment.Spec.Template.Spec.Containers[0].StartupProbe = nil
+
+	if alternateImageValue != "" {
+		log.Printf("setting debug image to %s\n", alternateImageValue)
+		deployment.Spec.Template.Spec.Containers[0].Image = alternateImageValue
+	}
+
+	fmt.Println("setting debug command to main container")
+
+	deployment.Spec.Template.Spec.Containers[0].Command = []string{"sleep", "infinity"}
+	deployment.Spec.Template.Spec.Containers[0].Args = []string{}
+
+	if err := updateDeployment(context, clusterNamespace, deployment); err != nil {
+		fmt.Printf("Failed to update deployment %s. %v\n", deployment.Name, err)
+		os.Exit(1)
+	}
+
+	deployment_Pod_Name, err := podRunningWithLabel(context, clusterNamespace, deployment.Spec)
+	if deployment_Pod_Name == "" || err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	if err := setDeploymentScale(context, clusterNamespace, deployment.Name, 0); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	fmt.Printf("waiting for the deployment pod %s to be deleted\n", deployment_Pod_Name)
+
+	err = waitForPodDeletion(context, clusterNamespace, deployment_Pod_Name)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	debug_deployment := &appsv1.Deployment{
```

Comment:
Vars are usually named with camelCase, otherwise it feels like bash :)
```suggestion
	debugDeployment := &appsv1.Deployment{
```

Reply:
done